lib/compat/wordpress-6.1/class-gutenberg-rest-edit-site-export-controller.php

```diff
@@ -0,0 +1,189 @@
+
+		$user_customisations = $this->get_user_customisations();
+
+		foreach ( $user_customisations['templates'] as $template ) {
+			$fileToUpdate = get_template_directory() . '/templates/' .
+				$template->slug . '.html';
+			if ( ! file_exists( $fileToUpdate ) ) {
+				throw new Exception("Theme file does not exist: " . $fileToUpdate);
+			}
+			if ( ! wp_is_writable( $fileToUpdate ) ) {
+				throw new Exception("Theme file is not writeable: " . $fileToUpdate);
+			}
+			file_put_contents(
+				$fileToUpdate,
+				$template->content
+			);
+		}
+
+		foreach ( $user_customisations['template_parts'] as $template_part ) {
+			$fileToUpdate = get_template_directory() . '/parts/' .
+				$template_part->slug . '.html';
+			if ( ! file_exists( $fileToUpdate ) ) {
+				throw new Exception("Theme file does not exist: " . $fileToUpdate);
+			}
+			if ( ! wp_is_writable( $fileToUpdate ) ) {
+				throw new Exception("Theme file is not writeable: " . $fileToUpdate);
+			}
+			file_put_contents(
+				$fileToUpdate,
+				$template_part->content
+			);
+		}
+	}
+
+	protected function update_theme_json_file() {
+		$fileToUpdate = get_template_directory() . '/theme.json';
+		if ( ! file_exists( $fileToUpdate ) ) {
+			throw new Exception("Theme file does not exist: " . $fileToUpdate);
+		}
+		if ( ! wp_is_writable( $fileToUpdate ) ) {
+			throw new Exception("Theme file is not writeable: " . $fileToUpdate);
+		}
+		file_put_contents(
+			$fileToUpdate,
+			gutenberg_export_theme_json()
+		);
+	}
+
+	protected function get_user_customisations() {
+
+		$templates = gutenberg_get_block_templates();
+		$template_parts = gutenberg_get_block_templates ( array(), 'wp_template_part' );
+
+		$exported_templates = [];
+		$exported_template_parts = [];
+
+		foreach ( $templates as $template ) {
+			if ( $template->source !== 'custom' ) {
+				continue;
+			}
+			$template = $this->clean_template( $template );
+			$exported_templates[] = $template;
+		}
+
+		foreach ( $template_parts as $template_part ) {
+			if ( $template_part->source !== 'custom' ) {
+				continue;
+			}
+			$template_part = $this->clean_template( $template_part );
+			$exported_template_parts[] = $template_part;
+		}
+
+		return array(
+			'templates'=>$exported_templates,
+			'template_parts'=>$exported_template_parts
+		);
+
+	}
+
+	/*
+	 * Clean a template or template part by reversing unicode escaping
+	 * and removing the theme attribute from the content
+	 */
+	function clean_template( $template ) {
+		$template->content = _remove_theme_attribute_in_block_template_content(
+			$template->content
+		);
+
+		// `serialize_block_attributes` has "unicode escape sequence
+ 		// substitution for characters which might otherwise interfere with embedding
+ 		// the result in an HTML comment".
+		// This reverses that escaping for saving back to files.
+		// @see `wp-includes/blocks.php`
+		$template->content = str_replace( '\\u002d\\u002d', '--', $template->content );
```

Comment:
We took the decision not to do this when exporting the theme. If we're happy doing it here then I think we should also do the same for export.

Reply:
Hmm the devex of working with code like this
```
 {"bottom":"var(\u002d\u002dwp\u002d\u002dcustom\u002d\u002dspacing\u002d\u002dsmall, 1.25rem)"}}}}
 ```
 is awful so we need to not do it.

Reply:
I agree, but if that's what we think we should make the same change in the export.